### PR TITLE
Too many open files error and high IO;

### PR DESCRIPTION
**Pull request by rguedes**
Short Description: 
Too many open files error and high IO

Fixes:
- Fix: "IOError: [Errno 24] Too many open files"

### CODE REVIEW

**Comment by Nihisil** [comment]
Hm, they should to be automatically closed, when you use `with open()`. Did you test it?

**Comment by rguedes** [comment]
Yes and the errors stopped

**Comment by DayBr3ak** [comment]
of course, `with` statement handle the closing of files automatically even when execeptions are raised. This pr should have no effect on the problem at all

**Comment by rguedes** [comment]
Apparently not close or not work well, at least in windows...

**Comment by Nihisil** [comment]
`with` will close files when GC will do the job. So, probably for some configuration it is too much time between GC runs, and for them we need to close files manually.

**Comment by DayBr3ak** [comment]
well if it its a GC problem, at least get rid of the with statement

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def update_web_location(self, cells=[], lat=None, lng=None, alt=None):
         user_data_cells = "data/cells-%s.json" % (self.config.username)
         with open(user_data_cells, 'w') as outfile:
             json.dump(cells, outfile)
+            outfile.close()
```
Comment (cwild) [inline pokemongo_bot/__init__.py]: ```with``` closes outfile automatically.

**Comment by DayBr3ak** [comment]
and to be fair I'm pretty sure with does't work with GC it's just sugar for a __enter__ and __exit__ method.

**Comment by makistsantekidis** [comment]
@Nihisil @DayBr3ak I think that when using a `with` statement the file is closed immediately after exiting the `with` context, since the __exit__ function of `open` is called which makes sure to close the file.

If you want you can confirm this by adding 

```
print outfile.closed
```
after the `with` contexts

**Comment by Surceis** [comment]
this https://github.com/PokemonGoF/PokemonGo-Bot/issues/1654

**Comment by douglascamata** [comment]
#1654 figures out the source of the problem, closing this and adding #1654 to the approved issue list.

**Pull request closed without merging**